src/Internal/Marshaller/Type/UuidType.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * This file is part of Temporal package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Temporal\Internal\Marshaller\Type;
+
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Temporal\Internal\Marshaller\MarshallingRule;
+use Temporal\Internal\Support\Inheritance;
+
+final class UuidType extends Type implements DetectableTypeInterface, RuleFactoryInterface
+{
+    public static function match(\ReflectionNamedType $type): bool
+    {
+        if (!\interface_exists(UuidInterface::class)) {
```

Comment:
It might be better to cached it in a static variable here

Reply:
Added static variable.